Add to_sentence helper

tommyschaefer opened the pull request:
Hi there!

This is my first pass at tackling #347 and implementing the `#to_sentence` text helper!

I added tests, the implementation, and some documentation (based on #275). I'm running into an error when I try and build the documentation to verify that I didn't mess up the markdown, though.

I ran `crystal docs` and got:

```
Unhandled type in `name`: Lucky::Action+:Class
Crystal::Doc::Type#name:String
Crystal::Doc::Type#full_name_without_type_vars<String::Builder>:String::Builder
Crystal::Doc::Type#full_name<String::Builder>:(String::Builder | Nil)
Crystal::Doc::Method#args_to_html<String::Builder, Bool>:(String::Builder | Nil)
Crystal::Doc::Method#id:String
Crystal::Doc::MethodSummaryTemplate#to_s<File>:File
Crystal::Doc::Generator#generate_types_docs<Array(Crystal::Doc::Type), String, Array(Crystal::Doc::Type)>:Nil
Crystal::Doc::Generator#generate_types_docs<Array(Crystal::Doc::Type), String, Array(Crystal::Doc::Type)>:Nil
Crystal::Command#run:(Bool | Crystal::Compiler::Result | IO::FileDescriptor | Nil)
main

Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

I'm not sure whether I should actually open an issue for this, or if I did something wrong 😄

Once everything is looking good to you all, I can squash the commits!

Thank you so much for your time and feedback!! 😄

hanneskaeufler commented:
Jepp `crystal docs` is broken (as in doesn't run on lucky right now), you're not doing anything wrong. To see your changes you can do something like `crystal docs path/to/something.cr` to just see docs for your one file.

tommyschaefer commented:
@hanneskaeufler Awesome! Thanks for the help with docs and nice spot on the typo! I appreciate the help!!

I've updated the method documentation to be a bit more readable after seeing what it looked like when rendered and squashed my commits. Let me know if there are any other improvements you all would like to see, and I'll be happy to make them! 😄

Thank you again for the feedback and help @hanneskaeufler!

hanneskaeufler commented on src/lucky/page_helpers/text_helpers.cr:
```diff
@@ -126,6 +126,54 @@ module Lucky::TextHelpers
+  #
+  # NOTE: By default `#to_sentence` will include a
+  # [serial comma](https://en.wikipedia.org/wiki/Serial_comma). This can be
+  # overriden like so:
```
I like the trivia about the oxford comma :) Not sure it warrants an extra `NOTE`, but also happy to see it here. `overriden => overridden`.

tommyschaefer replied:
Ah! Thanks for the catch of the misspelling! Sorry that it snuck through!

As far as the note is concerned, I noticed there was a [comment on the Rails documentation](https://apidock.com/rails/Array/to_sentence#1497-Grammatical-error) expressing some concern about the Oxford comma, so I figured it might be helpful to mention it. Do you think this is worth including? 😄 If so, do you think it should it be under a `NOTE` (I saw that `NOTE`s get handled specially by `crystal docs`)?

hanneskaeufler replied:
No worries about the typos, not much of a problem to have one sneak through :) I think the information per se is worth including, it makes the docs a bit more lively and interesting than just plain stating method documentation / arguments etc. Just my 2 cents :) Looking at the crystal api docs, the notes are more technically relevant I guess, so we should probably not overuse them, but 🤷‍♂️ I like it the more I think about it haha

paulcsmith replied:
I like it a lot! It's a nice thing to know and it is helpful to know how to override it when necessary
